controllers/networking-calico/pkg/controller/actuator.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"github.com/gardener/gardener-extensions/pkg/controller/network"
+	"github.com/gardener/gardener/pkg/chartrenderer"
+
+	calicov1aplha1 "github.com/gardener/gardener-extensions/controllers/networking-calico/pkg/apis/calico/v1alpha1"
+	"github.com/go-logr/logr"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	// StatusTypeMeta is the TypeMeta of Calico Status
+	StatusTypeMeta = metav1.TypeMeta{
+		APIVersion: calicov1aplha1.SchemeGroupVersion.String(),
+		Kind:       "NetworkConfigStatus",
+	}
+)
+
+type actuator struct {
+	logger logr.Logger
+
+	chartRenderer chartrenderer.Interface
+	restConfig    *rest.Config
+
+	client  client.Client
+	scheme  *runtime.Scheme
+	decoder runtime.Decoder
+}
+
+const LogID = "network-calico-actuator"
+
+// NewActuator creates a new Actuator that updates the status of the handled Network resources.
+func NewActuator() network.Actuator {
+	return &actuator{
+		logger: log.Log.WithName(LogID),
+	}
+}
+
+func (a *actuator) InjectScheme(scheme *runtime.Scheme) error {
+	a.scheme = scheme
+	a.decoder = serializer.NewCodecFactory(a.scheme).UniversalDecoder()
+	return nil
+}
+
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+func (a *actuator) InjectConfig(config *rest.Config) error {
+	a.restConfig = config
+
+	chartRenderer, err := chartrenderer.NewForConfig(config)
```

Comment:
This is a chart renderer for the seed cluster, however, you need a chart renderer for the shoot cluster.